File: Easy/520. Detect Capital/DetectCapital.py
'''
LeetCode Easy: 520. Detect Capital
https://leetcode.com/problems/detect-capital/

Check DetectCapital.cpp for brief solution explanation.
'''
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        N = len(word)
        capitals = 0
        for i in word:
            if i.isupper():
                capitals+=1
        if capitals==0 or capitals==N or (capitals==1 and word[0].isupper()):
            return True
        else:
            return False

File: Easy/520. Detect Capital/DetectCapital.java
/*
 * LeetCode Easy: 520. Detect Capital
 * https://leetcode.com/problems/detect-capital/
 * 
 * Check DetectCapital.cpp for brief solution explanation.
 */
class Solution {
    public boolean detectCapitalUse(String word) {
        int N = word.length();
        int capitals = 0;
        for (int i = 0; i < N; i++) {
            if (isCapital(word.charAt(i))) {
                capitals++;
            }
        }
        if (capitals == 0 || capitals == N || (capitals == 1 && isCapital(word.charAt(0)))) {
            return true;
        } else {
            return false;
        }
    }

    public boolean isCapital(char a) {
        return 'A' <= a && a <= 'Z';
    }
}